platform/mds/mds-web/src/main/resources/webapp/js/controllers.js

```diff
@@ -3865,13 +3890,35 @@
                 callback);
         };
 
-        $scope.setAvailableFieldsForDisplay = function() {
+        $scope.subclassCurrent = null;
+
+        $scope.setAvailableFieldsForDisplay = function(scope) {
             var i;
-            $scope.availableFieldsForDisplay = [];
-            for (i = 0; i < $scope.allEntityFields.length; i += 1) {
-                if (!$scope.allEntityFields[i].nonDisplayable) {
-                    $scope.availableFieldsForDisplay.push($scope.allEntityFields[i]);
+            var availableFieldsForDisplay = [];
+            for (i = 0; i < scope.allEntityFields.length; i += 1) {
+                if (!scope.allEntityFields[i].nonDisplayable) {
+                    availableFieldsForDisplay.push(scope.allEntityFields[i]);
+                }
+            }
+            var subclassnameCurrent = scope.$parent.$parent.$parent.subclassCurrent;
+            if(subclassnameCurrent) {
+                var entitiesT = scope.$parent.$parent.$parent.entitiesDerived;
+                var entitySelected = !entitiesT ? null : entitiesT.filter(function(e) {return e.name === subclassnameCurrent;})[0];
+                var fieldsAdded1 = !entitySelected? null : entitySelected.fieldsAdded;
+                var fieldsAdded = fieldsAdded1.slice(6);  // TODO: remove hack
+                if(fieldsAdded && fieldsAdded.length>0) {
+                    availableFieldsForDisplay = $scope.allEntityFields.concat(fieldsAdded);
```

Comment:
this code looks really suspect with all the hacks and $parent.$parent stuff. Moreover I'm not sure I understand what are we trying to do here. Are we adding fields from the child class to the parent view? Is that it?